indexer/lib/ipni-watcher.js

```diff
@@ -44,23 +44,7 @@ export async function getProvidersWithMetadata() {
   const res = await fetch('https://cid.contact/providers')
   assertOkResponse(res)
 
-  const providers = /**
-   * @type {{
-   *   AddrInfo: {
-   *     ID: string
-   *     Addrs: string[]
-   *   }
-   *   LastAdvertisement: {
-   *     '/': string
-   *   }
-   *   LastAdvertisementTime: string
-   *   Publisher: {
-   *     ID: string
-   *     Addrs: string[]
-   *   }
-   *   // Ignored: ExtendedProviders, FrozenAt
-   * }[]}
-   */ (await res.json())
```

Comment:
TypeScript does not recognise this syntax; it parses the type definition as a singular object, not as an array of objects. This is most likely a bug in [prettier-plugin-jsdoc](https://www.npmjs.com/package/prettier-plugin-jsdoc).

I am going to check if this can be fixed via `tsdoc: true` config - https://github.com/hosseinmd/prettier-plugin-jsdoc?tab=readme-ov-file#tsdoc

Reply:
Could we have a file `indexer/lib/ipni-watcher.d.ts`, just like there's C header and implementation files? I'm not the biggest fan of a central types definition. But if what I proposed isn't an option, I'd vote for moving the type to `typings.d.ts`

Reply:
AFAIK, you cannot have a `.d.ts` file with the same name as another `.js` file. How about creating a dedicated typing file for the IPNI API? That way we can eventually extract this typings file into a package we can share among many repositories.

See [f7ee4bb](https://github.com/CheckerNetwork/piece-indexer/pull/178/commits/f7ee4bba03941b7a6fd445e64a73f006bb2f3ad3)